handshake_ix.go

```diff
@@ -146,7 +146,7 @@ func ixHandshakeStage1(f *Interface, addr *udpAddr, hostinfo *HostInfo, packet [
 			return true
 		}
 
-		hostinfo, err = f.handshakeManager.AddIndex(myIndex, ci)
+		hostinfo, err = f.handshakeManager.AddIndex(vpnIP, myIndex, hs.Details.InitiatorIndex, ci)
```

Comment:
There's a lock call 9 lines below here that can probably be moved down above line 203:

```
203: hostinfo.HandshakePacket[0] = make([]byte, len(packet[HeaderLen:]))
```